processor/src/main/kotlin/com/thinkslynk/fabric/annotations/extensions/Iterable.kt

```diff
@@ -0,0 +1,25 @@
+package com.thinkslynk.fabric.annotations.extensions
+
+fun <T,S,R>Iterable<T>.crossFlatMap(extract:(T)->Iterable<S>, transform:(List<S>)->R):List<R>{
+    val result = mutableListOf<R>()
+    val each = map{extract(it).toList()}
+    if(each.any{it.isEmpty()})
+        return emptyList()
```

Comment:
This code produces a cartesian product and applies a `map` operation on that. If one of the inputs of a cartesian product is the empty set than you'd expect the result to be empty too. 

Atm where this code is called, I'm sure that no input list is ever empty but this is what I'd expect if I have a case where it could be. I can accept throwing an exception in case, but I can't accept filtering out the empty lists first.
If one of the inputs would be dynamic I wouldn't expect my input list of 5 elements to suddenly drop to only 4 cause 1 of my inputs is empty, without me being able to infer which index I'm missing.